src/error_handling_helpers.h

```diff
@@ -0,0 +1,181 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Note: migrated from rmw/error_handling.h in 2017-04
+
+#ifndef ERROR_HANDLING_HELPERS_H_
+#define ERROR_HANDLING_HELPERS_H_
+
+// When this define evaluates to true (default), then messages will printed to
+// stderr when an error is encountered while setting the error state.
+// For example, when memory cannot be allocated or a previous error state is
+// being overwritten.
+#ifndef RCUTILS_REPORT_ERROR_HANDLING_ERRORS
+#define RCUTILS_REPORT_ERROR_HANDLING_ERRORS 1
+#endif
+
+// When this define evaluates to true (default), then a warning will be written
+// to stderr using RCUTILS_SAFE_FWRITE_TO_STDERR.
+#ifndef RCUTILS_WARN_ON_TRUNCATION
+#define RCUTILS_WARN_ON_TRUNCATION 1
+#endif
+
+#ifndef __STDC_WANT_LIB_EXT1__
+#define __STDC_WANT_LIB_EXT1__ 1  // indicate we would like memmove_s if available
+#endif
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <rcutils/error_handling.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+// do not use externally, internal function which is only to be used by error_handling.c
+static
+size_t
+__rcutils_copy_string(char * dst, size_t dst_size, const char * src)
+{
+  assert(dst != NULL);
+  assert(dst_size > 0);
+  assert(src != NULL);
+  // doesn't matter how long src actually is if it is longer than dst, so limit to dst + 1
+  size_t src_length = strnlen(src, dst_size + 1);
```

Comment:
nit, since the check on line 60 includes `src_length` being equal to `dst_size`, I think the `+ 1` can be omitted here.

Reply:
Yeah, that's true, but I wanted to allow it to exceed (I was looking at it in lldb) and not just match. I think it can be omitted now though.